pkg/util/validation/validate.go

```diff
@@ -201,6 +200,20 @@ func ValidateSample(m *SampleValidationMetrics, now model.Time, cfg SampleValida
 	return nil
 }
 
+// ValidateSampleHistogram returns an err if the sample is invalid.
+// The returned error may retain the provided series labels.
+// It uses the passed 'now' time to measure the relative time of the sample.
+func ValidateSampleHistogram(m *SampleValidationMetrics, now model.Time, cfg SampleValidationConfig, userID, group string, ls []mimirpb.LabelAdapter, s mimirpb.Histogram) ValidationError {
+	unsafeMetricName, _ := extract.UnsafeMetricNameFromLabelAdapters(ls)
```

Comment:
Move this inside the `if` block, so that we read only if we have to return an error.

Reply:
[unrelated but nit] Could you do the same in `ValidateSample()`, please?

Reply:
doing both